.github/workflows/.test.yml

```diff
@@ -0,0 +1,483 @@
+# reusable workflow
+name: .test
+
+# TODO: hide reusable workflow from the UI. Tracked in https://github.com/community/community/discussions/12025
+
+on:
+  workflow_call:
+    inputs:
+      use_snapshotter:
+        required: true
+        type: boolean
+      storage_driver:
+        required: true
+        type: string
+
+env:
+  GO_VERSION: "1.20.7"
+  GOTESTLIST_VERSION: v0.3.1
+  TESTSTAT_VERSION: v0.1.3
+  ITG_CLI_MATRIX_SIZE: 6
+  DOCKER_EXPERIMENTAL: 1
+  DOCKER_GRAPHDRIVER: ${{ inputs.storage_driver }}
+  TEST_INTEGRATION_USE_SNAPSHOTTER: ${{ inputs.use_snapshotter && '1' || '' }}
+
+jobs:
+  validate-dco:
+    uses: ./.github/workflows/.dco.yml
+  build-dev:
+    runs-on: ubuntu-20.04
+    needs:
+      - validate-dco
+    strategy:
+      fail-fast: false
+      matrix:
+        mode:
+          - ""
+          - systemd
+    steps:
+      - name: Prepare
+        run: |
+          if [ "${{ matrix.mode }}" = "systemd" ]; then
+            echo "SYSTEMD=true" >> $GITHUB_ENV
+          fi
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build dev image
+        uses: docker/bake-action@v2
+        with:
+          targets: dev
+          set: |
+            *.cache-from=type=gha,scope=dev${{ matrix.mode }}
+            *.cache-to=type=gha,scope=dev${{ matrix.mode }},mode=max
+            *.output=type=cacheonly
+
+  validate-prepare:
+    runs-on: ubuntu-20.04
+    needs:
+      - validate-dco
+    outputs:
+      matrix: ${{ steps.scripts.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Create matrix
+        id: scripts
+        run: |
+          scripts=$(cd ./hack/validate && jq -nc '$ARGS.positional - ["all", "default", "dco"] | map(select(test("[.]")|not)) + ["generate-files"]' --args *)
+          echo "matrix=$scripts" >> $GITHUB_OUTPUT
+      - name: Show matrix
+        run: |
+          echo ${{ steps.scripts.outputs.matrix }}
+
+  validate:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 120
+    needs:
+      - validate-prepare
+      - build-dev
+    strategy:
+      fail-fast: true
+      matrix:
+        script: ${{ fromJson(needs.validate-prepare.outputs.matrix) }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up runner
+        uses: ./.github/actions/setup-runner
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build dev image
+        uses: docker/bake-action@v2
+        with:
+          targets: dev
+          set: |
+            dev.cache-from=type=gha,scope=dev
+      - name: Validate
+        run: |
+          make -o build validate-${{ matrix.script }}
```

Comment:
I think we could move validate jobs to a dedicated "validate" workflow like we do in buildkit: https://github.com/moby/buildkit/blob/master/.github/workflows/validate.yml